pymc4/distributions/continuous.py

```diff
@@ -1165,6 +1169,103 @@ def upper_limit(self):
         return self.conditions["high"]
 
 
+class Flat(ContinuousDistribution):
+    r"""A uniform distribution with support :math:`(-\inf, \inf)`.
+    Used as a uninformative log-likelihood that returns
+    zeros regardless of the passed values.
+    """
+
+    def __init__(self, name, **kwargs):
+        super().__init__(name, **kwargs)
+
+    @staticmethod
+    def _init_distribution(conditions):
+        return tfd.Uniform(low=-np.inf, high=np.inf)
+
+    def log_prob(self, value):
+        # convert the value to tensor
+        value = tf.convert_to_tensor(value)
+        # if not tf.rank(value):
+        #     value = tf.convert_to_tensor([value])
+        # value_shape = value.shape
+        # # if the rank of the `value` is less than distribution's `event shape` raise an error
+        # if tf.rank(value) < tf.rank(self._distribution.event_shape):
+        #     raise ValueError("Rank of input tensor less than distribution's event shape")
+        # # if the rightmost axis of `value` doesn't match the distribution's `event_shape`, raise an error
+        # if (
+        #     len(self._distribution.event_shape)
+        #     and value_shape[-1] != self._distribution.event_shape[-1]
+        # ):
+        #     raise ValueError(
+        #         "Event shape of input tensor not consistent with the distribution's event shape"
+        #     )
+        # # if the shape of `value` doesn't match the distribution's `batch_shape`, raise an error
+        # if (
+        #     len(self._distribution.batch_shape)
+        #     and value_shape[-2] != self._distribution.batch_shape[-1]
+        # ):
+        #     raise ValueError(
+        #         "Batch shape of input tensor not consistent with the distributions's batch shape"
+        #     )
+        if tf.rank(value) > 2:
```

Comment:
I have reduced sum when the shape of value if ``(n_samples, batch_shape, event_shape)`` along the ``event_shape`` axis. Shouldn't we reduce sum over ``n_samples`` axis?

Reply:
We only must reduce sum over the event shape axes. That's what truly distinguishes between the batch and event shapes. For example,

```python
>>> import tensorflow as tf
>>> from tensorflow_probability import distributions as tfd
>>> 
>>> d = tfd.Normal(loc=tf.zeros(1, 2), scale=1)
>>> d = tfd.Sample(d, sample_shape=(3, 4, 5))
>>> d.batch_shape.as_list()
[1, 2]
>>> d.event_shape.as_list()
[3, 4, 5]
>>> 
>>> x = d.sample(sample_shape=(6, 7))
>>> x.numpy().shape
(6, 7, 1, 2, 3, 4, 5)
>>> v = d.log_prob(x)
>>> v.numpy().shape
(6, 7, 1, 2)
```

Reply:
Oh! This is very clear now. Thanks!!